src/validation.cpp

```diff
@@ -3464,7 +3464,7 @@ static FILE* OpenDiskFile(const CDiskBlockPos &pos, const char *prefix, bool fRe
         return nullptr;
     fs::path path = GetBlockPosFilename(pos, prefix);
     fs::create_directories(path.parent_path());
-    FILE* file = fsbridge::fopen(path, "rb+");
```

Comment:
Isn't the following enough?
```cpp
FILE* file = fsbridge::fopen(path, fReadOnly ? "rb": "wb+");
if (!file) {
    LogPrintf("Unable to open file %s\n", path.string());
    return nullptr;
}
```
Why the second attempt when it's not read only?

Reply:
"w+" is not good because it truncates the file.
If we want to merge the two fopen, the best write mode is "ab+"

Reply:
Yes! Saw the manpage, but forgot to change after paste.